### DISCUSSION TITLE
Constructors

### DISCUSSION THREAD

**Discussion by yasharyan0904** [Cyfrin/foundry-full-course-cu#1867] [Q&A] [answered]
Can anyone just tell me why we need constructors and explain what it exactly does in a smart contract, in simple words!!

**Comment by EngrPips** [comment, marked as answer]
Hello @yasharyan0904, First of all, I want you to know that every contract has a constructor, whether you explicitly define it or not. A constructor is the function of a contract that is only executed once in the contract's lifetime. The constructor is responsible for initializing the runtime bytecode of a contract, and if you have any other variable that you want to initialize at the point of deployment, you explicitly specify it in the constructor. Please let me know If anything is still confusing.

**Reply by yasharyan0904** [reply]
So, in short 
1)Those variables that we want to initialize at the time of contract deployment we specify it under constructor!?
2)Generally these are those variables that we use for import contracts also!?

What if we don't specify it in constructor?? What will be it's effect on the variable??

**Reply by EngrPips** [reply]
Please explain what you mean by `Generally, these are those variables that we use for import contracts also!?` 

If you set a variable as immutable and did not initialize it in the constructor, your contract won't compile. You can experiment and let me know how it goes.

**Reply by yasharyan0904** [reply]
ok got it, Thanks:)
